src/ansys/mapdl/core/krylov.py

```diff
@@ -0,0 +1,612 @@
+            raise ValueError(
+                "The number of frequencies for which to compute the reduced solution is required to be greater than 0"
+            )
+        if not isinstance(load_key, int) or load_key < 0 or load_key > 1:
+            raise ValueError(
+                "The Load key value for computing the reduced solution is required to be 0 or 1"
+            )
+        if not isinstance(out_key, bool):
+            raise ValueError(
+                "The out_key value for computing the reduced solution is required to be Boolean True or False"
+            )
+
+    def _check_input_kryexpand(self, out_key, res_key):
+        """Validates the inputs to kryexpand method."""
+
+        if not isinstance(out_key, bool):
+            raise ValueError(
+                "The out_key value for expanding the reduced solution is required to be Boolean True or False"
+            )
+        if not isinstance(res_key, int) or res_key < 0 or res_key > 3:
+            raise ValueError(
+                "The res_key value for expanding the reduced solution is required to be 0 -> 3"
+            )
+
+    def _get_data_from_full_file(self):
+        """Extracts Stiffness, Mass, Damping and force matrices from full file"""
+
+        self._mat_k = self.mm.stiff(fname=self.full_file)
+        self._ndof = self._mat_k.shape[1]
+        self._mat_m = self.mm.mass(fname=self.full_file)
+        self._mat_c = self.mm.damp(fname=self.full_file)
+
+        self._mapdl.smat("Nod2Solv", "D", "IMPORT", "FULL", self.full_file, "NOD2SOLV")
+        self.Nod2Solv = self.mm.mat(name="Nod2Solv")
+
+        self._mapdl.vec("fz", "Z", "IMPORT", "FULL", self.full_file, "RHS")
+        self._mapdl.vec("fz0", "Z", "COPY", "fz")
+        self.fz0 = self.mm.vec(name="fz0")
+
+    def _chk_ortho_vec(self, orth_file, uz, num_q):
+        """Check Orthonormality of vectors"""
+
+        f = open(orth_file, "w")
+        for i in range(1, num_q + 1):
+            self._mapdl.vec("Vz", "Z", "LINK", self.Qz.id, i)
+            Vz = self.mm.vec(name="Vz")
+            Vz_m = Vz.asarray()
+            for j in range(1, num_q + 1):
+                self._mapdl.vec(uz.id, "Z", "LINK", self.Qz.id, j)
+                uz_m = uz.asarray()
+                rcon = np.vdot(Vz_m, uz_m)
+                rcon_real = rcon.real
+                rcon_imag = rcon.imag
+                dcon = np.sqrt(rcon_real * rcon_real + rcon_imag * rcon_imag)
+                f.write(f"V_{i}\tV_{j}\t{dcon}\n")
+        f.close()
+
+    def krygensub(
+        self, max_dim_q, freq_val, chk_ortho_key=False, out_key=False, full_file=None
+    ):
+        """Generates a Krylov subspace for model reduction in harmonic analysis
+
+        This method generates a Krylov subspace used for a model reduction
+        solution in a harmonic analysis. The subspace is built using the
+        assembled matrices and load vector on the jobname.full file that
```

Comment:
```suggestion
        assembled matrices and load vector on the ``<jobname>.full`` file that
```

Reply:
@PipKat I dont think there is a convention about that. But that one works for me!